### ISSUE TITLE
Inconsistent parse result for number list with non-plain text

### ISSUE THREAD

**Issue by fabon-f** [progfay/scrapbox-parser#1753]
**Describe the bug**

A parse result for number list with non-plain text is inconsistent to one with plain text.

```
 1. foo
 2. [* bar]
 3. `baz`
```

Scrapbox text above will be parsed as below:

```javascript
[
  {
    indent: 1,
    type: "line",
    nodes: [
      {
        type: "numberList",
        raw: "1. foo",
        rawNumber: "1",
        number: 1,
        nodes: [ [Object] ]
      }
    ]
  },
  {
    indent: 1,
    type: "line",
    nodes: [
      {
        type: "numberList",
        raw: "2. ",
        rawNumber: "2",
        number: 2,
        nodes: [] // no child nodes!
      },
      {
        type: "decoration",
        raw: "[* bar]",
        // etc.
      }
    ]
  },
  {
    indent: 1,
    type: "line",
    nodes: [
      {
        type: "numberList",
        raw: "3. ",
        rawNumber: "3",
        number: 3,
        nodes: [] // no child nodes!
      },
      { type: "code", raw: "`baz`", text: "baz" }
    ]
  }
]
```

2nd and 3rd `numberList` node doesn't have child nodes, while 1st has.

**To Reproduce**

```javascript
console.log(parse(
  ` 1. foo
 2. [* bar]
 3. \`baz\``,
  { hasTitle: false },
));
```

**Expected behavior**
All `numberList` node have appropriate children nodes.

**Comment by progfay** [comment]
I fixed this bug in #1755 

Please check diff of snapshots: 1d353d2514473b7a8b8826bbe6ac839e5008f03c

**Comment by progfay** [comment]
this issue was resolved in new version [`@progfay/scrapbox-parser@9.1.0`](https://www.npmjs.com/package/@progfay/scrapbox-parser/v/9.1.0)!
